configure.ac

```diff
@@ -15,16 +15,25 @@ AC_PREREQ([2.63])
 AC_INIT([kronosnet],
 	m4_esyscmd([build-aux/git-version-gen .tarball-version]),
 	[devel@lists.kronosnet.org])
+# Don't let AC_PROC_CC (invoked by AC_USE_SYSTEM_EXTENSIONS) replace
+# undefined CFLAGS with -g -O2, overriding our special OPT_CFLAGS.
+: ${CFLAGS=""}
 AC_USE_SYSTEM_EXTENSIONS
 AM_INIT_AUTOMAKE([1.13 dist-bzip2 dist-xz color-tests -Wno-portability subdir-objects])
 
-# Switch to RUNPATH by default and disable RPATH
-# it is necessary to have this done very early in ./configure
-# before libtool does linker detection.
-# see also: https://github.com/kronosnet/kronosnet/issues/107
-LDFLAGS="$LDFLAGS -Wl,--enable-new-dtags"
 LT_PREREQ([2.2.6])
+# --enable-new-dtags: Use RUNPATH instead of RPATH.
+# It is necessary to have this done before libtool does linker detection.
+# See also: https://github.com/kronosnet/kronosnet/issues/107
+# --as-needed: Modern systems have builtin ceil() making -lm superfluous but
+# AC_SEARCH_LIBS can't detect this because it tests with a false prototype
+AX_CHECK_LINK_FLAG([-Wl,--enable-new-dtags],[AM_LDFLAGS=-Wl,--enable-new-dtags])
+AX_CHECK_LINK_FLAG([-Wl,--as-needed],[AM_LDFLAGS="$AM_LDFLAGS -Wl,--as-needed"])
+AC_SUBST([AM_LDFLAGS])
```

Comment:
quick question here. does AX_CHECK_LINK_FLAG fails if the flag is not supported? right now we depend on this flag being available and working on several distributions. I am happy with the whole patch, but I would prefer to see configure fail if the flag is not available. I also had another item in my TODO list here: https://trello.com/c/TlsPgqgE/1433-configureac-hardening to have configure fails early if we don´t find some of the build deps etc.

Reply:
No, it does not fail. I didn't think either flag was really critical. `--enable-new-dtags` is required for proper module load order in the build tests, but does not affect the core library functionality after install. `--as-needed` gets rid of unneeded dependencies, but there aren't too many anyway, so this is a minor performance optimization only.

Reply:
BTW have you got concrete examples of failures `configure` should fail on but it does not? Where is it not "hard" enough currently? I haven't checked it with this in mind. It probably does some superfluous checks as well.

Reply:
for example we check for some binaries but if they are not there we don´t fail (Doxygen for example), but the idea was to have configure fails as early as possible if checks are not passing, I didn´t review much but the lack of Doxygen on the builders triggered the idea.

Reply:
As for the flags, I think at least new-dtags should cause a failure. we don´t know what the end user is doing on the system (production? development?) and either ways we should be able to function properly without knowing :-) agreed on as-needed, that´s optional.